### ISSUE TITLE
How to log cur_val_loss to tensorboard in validation_step?

### ISSUE THREAD

**Issue by RafailFridman** [PyTorchLightning/pytorch-lightning#1127]
I tried to get it to tensorboard this way
```python
def validation_step(self, batch, batch_nb):
        data, target= batch
        output = self.forward(data)
        val_loss = nn.MSELoss(reduction = "mean")(output, target)
        return {'val_loss': val_loss, 'log':{'cur_val_loss':val_loss}}
```
but that didn't work.
How can I monitor validation loss not only at the end of each epoch, but during validation?

**Comment by Borda** [comment]
@jeffling @awaelchli pls ^^

**Comment by awaelchli** [comment]
I have never come across a situation where monitoring the validation loss this way is meaningful. What is your use case here? I am curious. 
The reason why I say this is because the loggers are not really designed to do what you ask for (please correct me if I'm wrong). The loggers have a step that monotonically increases, it can mean training steps, epochs, or something else. But since the validation step is reset every epoch, it wouldn't make sense to use the same logger for training and validation. So I have don't have an answer for you, sorry.

Is it not enough in your case to compute the mean or maybe the std for the validation loss?

**Comment by williamFalcon** [comment]
you can just call the logger yourself whenever... i assume it’s bc the val set is huge or something? either way we normally track the avg at the end of the val epoch. but if you need to do it intro epoch then 
self.logger.do_whatever_tensorboard_does

**Comment by epignatelli** [comment]
Is there a counter that keeps track of the validation steps already performed? Or using `self.global_step`?
`self.logger.experiment.add_scalar("val_loss", val_loss, self.global_step)`, or ..?

**Comment by williamFalcon** [comment]
yes! if you wait a day we’ll post docs on the new simple way of doing it :)

**Comment by epignatelli** [comment]
Oh, thanks @williamFalcon, that's great! Is there a specific pr I can scrabble through in the meanwhile?
